js/service_worker.js

```diff
@@ -0,0 +1,21 @@
+self.addEventListener('activate', function (event) {
+  console.log('Service Worker: claiming control...');
+  return self.clients.claim();
+});
+
+self.addEventListener('fetch', function (event) {
+  if (event.request.destination === "image" && new URL(event.request.url).pathname.startsWith('/cantaloupe/iiif/') && new URL(location).searchParams.has('token')) {
```

Comment:
Not sure the `.startsWith('/cantaloupe/iiif')` bit here is properly portable? Is into deployment details?